vendors/cypress/boards/CYW943907AEVAL1F/ports/wifi/iot_wifi.c

```diff
@@ -492,14 +533,35 @@ WIFIReturnCode_t WIFI_GetPMMode( WIFIPMMode_t * pxPMModeType,
 }
 /*-----------------------------------------------------------*/
 
-BaseType_t WIFI_IsConnected(void)
+BaseType_t WIFI_IsConnected( const WIFINetworkParams_t * pxNetworkParams )
 {
     /* FIX ME. */
     configPRINTF( ( "WIFI_IsConnected...\r\n") );
+
+    BaseType_t xIsConnected = pdFALSE;
+
+    if ( isConnected )
+    {
+        if ( pxNetworkParams )
+        {
+            if ( pxNetworkParams->ucSSIDLength > 0
+                 && pxNetworkParams->ucSSIDLength <= wificonfigMAX_SSID_LEN
+                 && memcmp( pxNetworkParams->ucSSID, pcConnectedSSID, pxNetworkParams->ucSSIDLength)
+                 && pxNetworkParams->ucSSIDLength == strlen( pcConnectedSSID ) )
+            {
+                xIsConnected = pdTRUE;
+            }
+        }
+        else
+        {
+            xIsConnected = pdTRUE;
+        }
+    }
+
     return isConnected;
```

Comment:
Seems it needs to return `xIsConnected` here

Reply:
Thanks for catching this. Prepared the fix in local commit. Pushing shortly with other changes per review.